spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionServiceTests.java

```diff
@@ -372,6 +377,52 @@ public void testRestoreAppPropertiesV2() throws IOException {
 			verify(this.taskLauncher, never()).destroy(TASK_NAME_ORIG);
 		}
 
+		@Test
+		@DirtiesContext
+		public void testSavesRequestedVersionNoLabel() throws IOException {
+			initializeMultiVersionRegistry(appRegistry);
+
+			when(taskLauncher.launch(any())).thenReturn("0", "1");
+
+			Map<String, String> properties = new HashMap<>(1);
+			properties.put("version.timestamp", "1.0.1");
+
+			long firstTaskExecutionId = this.taskExecutionService.executeTask("t1", properties, new LinkedList<>());
+			this.taskRepository.completeTaskExecution(firstTaskExecutionId, 0, new Date(), "all done");
+
+			TaskManifest lastManifest = this.dataflowTaskExecutionMetadataDao.getLatestManifest("t1");
+
+			assertEquals("file:src/test/resources/apps/foo-task101", lastManifest.getTaskDeploymentRequest().getResource().getURL().toString());
+			assertEquals("default", lastManifest.getPlatformName());
+			assertEquals(1, lastManifest.getTaskDeploymentRequest().getDeploymentProperties().size());
+			assertEquals("1.0.1", lastManifest.getTaskDeploymentRequest().getDeploymentProperties().get("version.timestamp"));
+
+			verify(this.taskLauncher, never()).destroy(TASK_NAME_ORIG);
+		}
+
+		@Test
+		@DirtiesContext
+		public void testSavesRequestedVersionLabel() throws IOException {
```

Comment:
may want to add a negative test where they ask for a version that does not exist.

Reply:
Prob a follow up issue. Not sure if we do fail fast even on a stream side and would eventually result a runtime failure.